go/vt/vtgate/executor_dml_test.go

```diff
@@ -943,6 +943,23 @@ func TestInsertOnDupKey(t *testing.T) {
 	}
 }
 
+func TestAutocommitFail(t *testing.T) {
+	executor, sbc1, _, _ := createExecutorEnv()
+
+	query := "insert into user (id) values (1)"
+	sbc1.MustFailCodes[vtrpcpb.Code_ALREADY_EXISTS] = 1
+	masterSession.Reset()
```

Comment:
The next time you touch this file, it might be a good idea to fix all tests to defer a call to masterSession.Reset(). That should fix the race we saw with this test AND prevent it for any future tests added to this suite.